openchs-server-api/src/main/java/org/openchs/service/EncounterService.java

```diff
@@ -0,0 +1,88 @@
+        this.encounterRepository = encounterRepository;
+        this.individualService = individualService;
+        this.individualRepository = individualRepository;
+    }
+
+    public EncounterContract getEncounterByUuid(String uuid) {
+        Encounter encounter = encounterRepository.findByUuid(uuid);
+        return constructEncounters(encounter);
+    }
+
+    public Page<EncounterContract> getAllCompletedEncounters(String uuid, String encounterTypeUuids, DateTime encounterDateTime, DateTime earliestVisitDateTime, Pageable pageable){
+        Page<EncounterContract> encountersContract = null;
+        List<String> encounterTypeIdList = new ArrayList<>();
+        if(encounterTypeUuids != null) {
+            encounterTypeIdList = Arrays.asList(encounterTypeUuids.split(","));
+        }
+        Individual individual = individualRepository.findByUuid(uuid);
+        encountersContract = encounterRepository.findAll(
+                where(encounterRepository.withProgramEncounterId(individual.getId()))
+                        .and(encounterRepository.withProgramEncounterTypeIdUuids(encounterTypeIdList))
+                        .and(encounterRepository.withProgramEncounterEarliestVisitDateTime(earliestVisitDateTime))
+                        .and(encounterRepository.withProgramEncounterDateTime(encounterDateTime))
+                        .and(encounterRepository.withNotNullEncounterDateTime())
+                ,pageable).map(encounter -> constructEncounters(encounter));
+        return encountersContract;
+    }
+
+    public EncounterContract constructEncounters(Encounter encounter) {
+            EncounterContract encountersContract = new EncounterContract();
+            EncounterTypeContract encounterTypeContract = new EncounterTypeContract();
+            encounterTypeContract.setName(encounter.getEncounterType().getName());
+            encounterTypeContract.setUuid(encounter.getEncounterType().getUuid());
+            encounterTypeContract.setEncounterEligibilityCheckRule(encounter.getEncounterType().getEncounterEligibilityCheckRule());
+            encountersContract.setUuid(encounter.getUuid());
+            encountersContract.setName(encounter.getName());
+            encountersContract.setEncounterType(encounterTypeContract);
+            encountersContract.setSubjectUUID(encounter.getIndividual().getUuid());
+            encountersContract.setEncounterDateTime(encounter.getEncounterDateTime());
+            encountersContract.setCancelDateTime(encounter.getCancelDateTime());
+            encountersContract.setEarliestVisitDateTime(encounter.getEarliestVisitDateTime());
+            encountersContract.setMaxVisitDateTime(encounter.getMaxVisitDateTime());
+            encountersContract.setVoided(encounter.isVoided());
+            if(encounter.getObservations() != null) {
+                encountersContract.setObservations(individualService.constructObservations(encounter.getObservations()));
```

Comment:
Should not we extract out constructObservations into ObservationsService since it has nothing to do with Indvidual itself?